File: app/src/main/java/org/skyfaced/mvp/di/PresenterModule.java
package org.skyfaced.mvp.di;

import org.skyfaced.mvp.service.instant.InstantService;
import org.skyfaced.mvp.service.network.WaifuRepository;
import org.skyfaced.mvp.ui.MainPresenter;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public final class PresenterModule {
    @Provides
    @Singleton
    MainPresenter mainPresenter(WaifuRepository waifuRepository, InstantService instantService) {
        return new MainPresenter(waifuRepository, instantService);
    }
}

File: app/src/main/java/org/skyfaced/mvp/di/PresenterModule.kt
package org.skyfaced.mvp.di

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import org.skyfaced.mvp.service.instant.InstantService
import org.skyfaced.mvp.service.network.WaifuRepository
import org.skyfaced.mvp.ui.MainPresenter
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object PresenterModule {
    @Provides
    @Singleton
    fun mainPresenter(
        waifuRepository: WaifuRepository,
        instantService: InstantService,
    ) = MainPresenter(waifuRepository, instantService)
}